### PR TITLE
 add LoadingTableViewDataSource tests

### PR DESCRIPTION
**Pull request by imJustik**
*  remove dequeueReusable methods from UITableView extension
*  add LoadingTableViewDataSource tests

### CODE REVIEW

File: YARCH/Helpers/Extensions/UITableViewExtention.swift [hunk 1]
```diff
@@ -1,11 +1,7 @@
 import UIKit
 
 extension UITableView {
-    
-    static func defaultReuseId(for elementType: UIView.Type) -> String {
-        return String(describing: elementType)
-    }
-    
+
     convenience init(delegate: UITableViewDelegate, dataSource: UITableViewDataSource, backgroundColor: UIColor = UIColor.white, estimatedRowHeight: CGFloat = 80) {
```
Comment (khramtsoff) [inline YARCH/Helpers/Extensions/UITableViewExtention.swift:5]: UIColor.white - use type omission

File: YARCH/Sources/LoadingTableView/LoadingTableViewDataSource.swift [hunk 2]
```diff
@@ -9,21 +9,22 @@ extension LoadingTableViewDataSource {
...
     }
 
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return appearance.numberOfRowsInSection
+        return configuration.numberOfRowsInSection
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        return tableView.dequeueReusableCellWithRegistration(type: LoadingTableViewCell.self, indexPath: indexPath)
+        tableView.register(LoadingTableViewCell.self, forCellReuseIdentifier: String(describing: LoadingTableViewCell.self))
```
Comment (khramtsoff) [inline YARCH/Sources/LoadingTableView/LoadingTableViewDataSource.swift:27]: performance issue

File: YARCH/Sources/LoadingTableView/LoadingTableViewDelegate.swift [hunk 3]
```diff
@@ -15,7 +15,8 @@ class LoadingTableViewDelegate: NSObject, UITableViewDelegate {
     }
 
     func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-       return tableView.dequeueReusableHeaderFooterWithRegistration(type: LoadingSectionHeaderView.self)
+        tableView.register(LoadingSectionHeaderView.self, forHeaderFooterViewReuseIdentifier: String(describing: LoadingSectionHeaderView.self))
```
Comment (khramtsoff) [inline YARCH/Sources/LoadingTableView/LoadingTableViewDelegate.swift:18]: performance issue

File: YARCH/Sources/Modules/Catalog/CatalogTableDataSource.swift [hunk 4]
```diff
@@ -12,9 +12,11 @@ class CatalogTableDataSource: NSObject, UITableViewDataSource {
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCellWithRegistration(type: CatalogCell.self, indexPath: indexPath)
-        guard let viewModel = representableViewModels[safe: indexPath.row] else { return cell }
-        cell.configure(cellRepresentable: viewModel)
+        tableView.register(CatalogCell.self, forCellReuseIdentifier: String(describing: CatalogCell.self))
```
Comment (khramtsoff) [inline YARCH/Sources/Modules/Catalog/CatalogTableDataSource.swift:15]: performance issue

**Review by iltsarev — DISMISSED** [review]

File: YARCH/Sources/LoadingTableView/Tests/LoadingTableViewDataSourceTests.swift [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            it("should return number of rows in section") {
+                //when
+                let numberOfRowsInSection = dataSource.tableView(tableView, numberOfRowsInSection: 0)
+                //then
+                expect(numberOfRowsInSection).to(equal(TestData.configuration.numberOfRowsInSection))
+            }
+        }
+
+        describe(".cellForRowAtIndexPath") {
+            it("shoud register cell class") {
```
Comment (xzeror) [inline YARCH/Sources/LoadingTableView/Tests/LoadingTableViewDataSourceTests.swift]: typo: shoud -> should

File: YARCH/Sources/LoadingTableView/Tests/LoadingTableViewDelegateTests.swift [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                _ = delegate.tableView(tableView, viewForHeaderInSection: 0)
+
+                //then
+                expect(tableView.dequeueReusableHeaderFooterDidCalled).to(equal(1))
+                expect(tableView.dequeueReusableHeaderFooterArguments).to(equal(expectedHeaderIdentifier))
+            }
+        }
+
+        describe(".heightForHeaderInSection") {
+            it("shoud return height of section") {
```
Comment (xzeror) [inline YARCH/Sources/LoadingTableView/Tests/LoadingTableViewDelegateTests.swift]: typo: shoud -> should

Comment (xzeror) [reply YARCH/Sources/LoadingTableView/Tests/LoadingTableViewDelegateTests.swift]: I believe it must be "should return height of section header"

File: YARCH/Sources/Modules/Catalog/CatalogTableDataSource.swift [hunk 7]
```diff
@@ -12,9 +12,11 @@ class CatalogTableDataSource: NSObject, UITableViewDataSource {
...
+        let cell = tableView.dequeueReusableCell(withIdentifier: String(describing: CatalogCell.self), for: indexPath)
+        guard let viewModel = representableViewModels[safe: indexPath.row],
```
Comment (xzeror) [inline YARCH/Sources/Modules/Catalog/CatalogTableDataSource.swift]: Using guard here brings duplication of return statement. I think that using if statement will look more cleaner. Compare:
```
if let viewModel = representableViewModels[safe: indexPath.row],
   let catalogCell = cell as? CatalogCell {
    catalogCell.configure(cellRepresentable: viewModel)
}
return cell
```

**Review by xzeror — DISMISSED** [review]
Needs some minor fixes.

**Review by khramtsoff — CHANGES_REQUESTED** [review]
Remove `tableView.register` from `cellForRow`, `viewForHeaderInSection`; apply type omission.

File: YARCH/Sources/LoadingTableView/LoadingTableViewDataSource.swift (hunk 2 above)
Comment (imJustik) [reply YARCH/Sources/LoadingTableView/LoadingTableViewDataSource.swift:27]: Nope

File: YARCH/Sources/LoadingTableView/LoadingTableViewDelegate.swift (hunk 3 above)
Comment (imJustik) [reply YARCH/Sources/LoadingTableView/LoadingTableViewDelegate.swift:18]: Nope

File: YARCH/Sources/Modules/Catalog/CatalogTableDataSource.swift (hunk 4 above)
Comment (imJustik) [reply YARCH/Sources/Modules/Catalog/CatalogTableDataSource.swift:15]: Nope

**Pull request closed without merging**